contentgrid-spring-data-pagination/src/test/java/com/contentgrid/spring/data/pagination/jpa/strategy/TimedDirectCountItemCountStrategyTest.java

```diff
@@ -0,0 +1,125 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.transaction.PlatformTransactionManager;
+import org.springframework.transaction.TransactionDefinition;
+
+@SpringBootTest(
+        classes = InvoicingApplication.class,
+        properties = {
+                "spring.datasource.hikari.connection-init-sql=set search_path=intercept,public"
+        })
+class TimedDirectCountItemCountStrategyTest {
+
+    @BeforeEach
+    void seedCustomers() {
+        customerRepository.saveAllAndFlush(
+                IntStream.range(0, 25)
+                        .mapToObj(number -> {
+                            var customer = new Customer();
+                            customer.setName("Customer %d".formatted(number));
+                            customer.setVat("VAT%d".formatted(number));
+                            return customer;
+                        })
+                        .toList()
+        );
+    }
+
+    @AfterEach
+    void destroyCustomers() throws SQLException {
+        // Ensure that interception schema is cleaned up.
+        // This MUST run outside of a transaction used for tests, because otherwise the query blocks
+        // because the open transaction has objects in this schema in use
+        try (var conn = dataSource.getConnection()) {
+            try (var statement = conn.createStatement()) {
+                statement.execute("DROP SCHEMA IF EXISTS intercept CASCADE;");
+            }
+        }
+
+        // Clean up created customer objects
+        customerRepository.deleteAll();
+    }
+
+    @Autowired
+    TimedDirectCountItemCountStrategy timedDirectCountCountingStrategy;
+
+    @Autowired
+    DataSource dataSource;
+
+    @Autowired
+    EntityManager entityManager;
+
+    @Autowired
+    PlatformTransactionManager transactionManager;
+
+    @Autowired
+    CustomerRepository customerRepository;
+
+    @Test
+    void performCount_withinTime() {
+        // Test needs to run in a transaction to have an open entity manager to perform queries
+        var tx = transactionManager.getTransaction(TransactionDefinition.withDefaults());
+        try {
+            Supplier<JPQLQuery<?>> querySupplier = () -> new JPAQuery<>(entityManager)
+                    .from(QCustomer.customer)
+                    .select(QCustomer.customer);
+
+            assertThat(timedDirectCountCountingStrategy.countQuery(querySupplier))
+                    .hasValueSatisfying(result -> {
+                        assertThat(result.isEstimated()).isFalse();
+                        assertThat(result.count()).isEqualTo(25);
+                    });
+        } finally {
+            transactionManager.commit(tx);
+        }
+    }
+
+    @Test
+    void performCount_outOfTime() throws SQLException {
+        try (var conn = dataSource.getConnection()) {
```

Comment:
This might need a bit of explanation

Reply:
This line, or the view below? And how much do I have explain? 

- The search path from the init-sql; how this view gets put inbetween
- what a `cross join lateral` is
- how the sleep works

The part that I think is obvious is that it makes the query slower.
Is it necessary to explain all the details of how it works for some test code?